numpy/ma/core.py

```diff
@@ -3050,15 +3050,18 @@ def __getitem__(self, indx):
         # mask of being reshaped if it hasn't been set up properly yet...
         # So it's easier to stick to the current version
         _mask = self._mask
-        if not getattr(dout, 'ndim', False):
+        # Did we extract a single item?  Note that if self contains an object
+        # array of arrays (yes, that happens), then dout will have .ndim.
+        if(not getattr(dout, 'ndim', False) or
+           (self.dtype.type is np.object_ and self.dtype is not dout.dtype)):
             # A record ................
             if isinstance(dout, np.void):
-                mask = _mask[indx]
+                mask = nomask if _mask is nomask else _mask[indx]
```

Comment:
That's right -- did that in the process and forgot about it...

I'll add a test...

(BTW, shall I volunteer you for maintainer........)